doc/manual/src/protocols/store-path.md

```diff
@@ -0,0 +1,126 @@
+- ```ebnf
+  fingerprint = type ":" sha256 ":" inner-digest ":" store ":" name
+  ```
+
+  Note that it includes the location of the store as well as the name to make sure that changes to either of those are reflected in the hash
+  (e.g. you won't get `/nix/store/<digest>-name1` and `/nix/store/<digest>-name2`, or `/gnu/store/<digest>-name1`, with equal hash parts).
+
+- `type` = one of:
+
+  - ```ebnf
+    | "text" ( ":" store-path )*
+    ```
+
+    for encoded derivations written to the store.
+    The optional trailing store paths are the references of the store object.
+
+  - ```ebnf
+    | "source" ( ":" store-path )*
+    ```
+
+    For paths copied to the store and hashed via a [Nix Archive (NAR)] and [SHA-256][sha-256].
+    Just like in the text case, we can have the store objects referenced by their paths.
+    Additionally, we can have an optional `:self` label to denote self reference.
+
+  - ```ebnf
+    | "output:" id
+    ```
+
+    For either the outputs built from derivations,
+    paths copied to the store hashed that area single file hashed directly, or the via a hash algorithm other than [SHA-256][sha-256].
+    (in that case "source" is used; this is only necessary for compatibility).
+
+    `id` is the name of the output (usually, "out").
+    For content-addressed store objects, `id`, is always "out".
+
+- `inner-digest` = base-16 representation of a SHA-256 hash of `inner-fingerprint`
+
+## Inner fingerprint
+
+- `inner-fingerprint` = one of the following based on `type`:
+
+  - if `type` = `"text:" ...`:
+
+    the string written to the resulting store path.
+
+  - if `type` = `"source:" ...`:
+
+    the the hash of the [Nix Archive (NAR)] serialization of the [file system object](@docroot@/store/file-system-object.md) of the store object.
+
+  - if `type` = `"output:" id`:
+
+    - For input-addressed derivation outputs:
+
+      the [ATerm](@docroot@/protocols/derivation-aterm.md) serialization of the derivation modulo fixed output derivations.
+
+    - For content-addressed store paths:
+
+      ```ebnf
+      "fixed:out:" rec algo ":" hash ":"
+      ```
+
+      where
+
+      - `rec` = one of:
+
+        - ```ebnf
+          | "r:"
+          ```
+          hashes of the for [Nix Archive (NAR)] (arbitrary file system object) serialization
+
+        - ```ebnf
+          |
+          ```
+          (empty string) for hashes of the flat (single file) serialization
+
+      - ```ebf
+        algo = "md5" | "sha1" | "sha256"
+        ```
+
+      - `hash` = base-16 representation of the path or flat hash of the contents of the path (or expected contents of the path for fixed-output derivations).
+
+      Note that `id` = `"out"`, regardless of the name part of the store path.
+      Also note that NAR + SHA-256 must not use this case, and instead must use the `type` = `"source:" ...` case.
+
+[Nix Archive (NAR)]: @docroot@/glossary.md#gloss-NAR
+[sha-256]: https://en.m.wikipedia.org/wiki/SHA-256
+
+## Historical Note
```

Comment:
```suggestion
### Historical Note
```

This is a note about the inner fingerprint, so let's make it a subsection of it.